nixos/modules/services/blockchain/nimbus/beacon-node.nix

```diff
@@ -0,0 +1,236 @@
+          example = "extip:12.34.56.78";
+          description = literalMD ''
+            Method for determining public address. (any, none, upnp, pmp, extip:<IP>)
+          '';
+        };
+
+        metrics = {
+          enable = lib.mkEnableOption (mdDoc "Nimbus beacon node metrics endpoint");
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Metrics address for beacon node.";
+          };
+          port = mkOption {
+            type = types.port;
+            default = 9100;
+            description = mdDoc "Metrics port for beacon node.";
+          };
+        };
+
+        rest = {
+          enable = lib.mkEnableOption (mdDoc "Nimbus beacon node REST API");
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Listening address of the REST API server.";
+          };
+
+          port = mkOption {
+            type = types.port;
+            default = 5052;
+            description = mdDoc "Port for the REST API server.";
+          };
+        };
+
+        extraArgs = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          example = ["--num-threads=1" "--graffiti=1337_h4x0r"];
+          description = mdDoc "Additional arguments passed to node.";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = length cfg.web3Urls > 0;
+        message = "Nimbus beacon node requires at least one Web3 URL to work!"; }
+    ];
+
+    users.users = optionalAttrs (cfg.service.user == "nimbus") {
+      nimbus = {
+        group = cfg.service.group;
+        home = cfg.dataDir;
+        description = "Nimbus beacon node service user";
+        isSystemUser = true;
+      };
+    };
+
+    users.groups = optionalAttrs (cfg.service.user == "nimbus") {
+      nimbus = { };
+    };
+
+    systemd.services.nimbus-beacon-node = {
+      description = "Nimbus Beacon Node (Ethereum consensus client)";
+
+      serviceConfig = mkMerge [{
+        User = cfg.service.user;
+        Group = cfg.service.group;
+
+        ExecStart = concatStringsSep " \\\n  " ([
+          "${cfg.package}/bin/nimbus_beacon_node"
+          "--network=${cfg.network}"
+          "--data-dir=${cfg.dataDir}"
+          "--log-level=${toUpper cfg.log.level}"
+          "--log-format=${cfg.log.format}"
+          "--nat=${cfg.nat}"
+          "--tcp-port=${toString cfg.listenPort}"
+          "--udp-port=${toString cfg.discoverPort}"
+          "--subscribe-all-subnets=${boolToString cfg.subAllSubnets}"
+          "--doppelganger-detection=${boolToString cfg.doppelganger}"
+          "--rest=${boolToString cfg.rest.enable}"
+        ] ++ optionals cfg.rest.enable [
+          "--rest-address=${cfg.rest.address}"
+          "--rest-port=${toString cfg.rest.port}"
+        ] ++ [
+          "--metrics=${boolToString cfg.metrics.enable}"
```

Comment:
```suggestion
          "--rest=${boolToString cfg.rest.enable}"
          "--metrics=${boolToString cfg.metrics.enable}"
        ] ++ optionals cfg.rest.enable [
          "--rest-address=${cfg.rest.address}"
          "--rest-port=${toString cfg.rest.port}"
```

Reply:
We should group the flags together that are unconditional. Having some between the conditional ones is confusing.

Reply:
It is more confusing if conditional and none conditional flags are mixed.

Reply:
No, in my opinion it's more confusing if you have flags with different prefixes mixed rather than grouped.

But if you intend to block this PR on this issue then I will change it.